src/app.jsx

```diff
@@ -324,12 +324,24 @@ class Application extends React.Component {
         cockpit.script("echo $XDG_RUNTIME_DIR")
                 .done(xrd => {
                     sessionStorage.setItem('XDG_RUNTIME_DIR', xrd.trim());
-                    this.init(false);
+                    if (!utils.isRootUser()) {
+                        this.init(false);
+                    } else {
+                        this.setState({
+                            userImagesLoaded: true,
+                            userContainersLoaded: true,
+                            userServiceAvailable: false
+                        });
+                    }
                 })
                 .fail(e => console.log("Could not read $XDG_RUNTIME_DIR: ", e.message));
     }
 
     checkUserService() {
+        if (utils.isRootUser()) {
```

Comment:
Please let's not repeat that check. Let's rather move the call of this inside the `XDG_RUNTIME_DIR` result processing above, when we already know the answer.